server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -628,10 +631,7 @@ private void parseValueAddFields(ParseContext context, String value, String fiel
             }
 
             if (fieldType().hasDocValues()) {
-                if (context.doc().getField(fieldType().name()) == null || !context.doc().getFields(fieldType().name()).equals(field)) {
-                    if (fieldName.equals(fieldType().name())) {
-                        context.doc().add(new SortedSetDocValuesField(fieldType().name(), binaryValue));
-                    }
+                if (context.doc().getField(fieldType().name()) == null || !context.doc().getField(fieldType().name()).equals(field)) {
```

Comment:
Super minor, I think you could simplify it a bit with:

```suggestion
                if (!Objects.equals(context.doc().getField(fieldType().name()), field)) {
```